### PR TITLE
StringTensor updates.

### PR DESCRIPTION
**Pull request by dan-zheng**
StringTensor updates based on upcoming changes to tensorflow `master` branch:

> Eliminate unnecessary packing/unpacking of tstring tensors across the C-API.

Code authored by @gharibian.

### CODE REVIEW

File: Sources/TensorFlow/Core/StringTensor.swift [hunk 1]
```diff
@@ -46,50 +46,72 @@ extension StringTensor {
...
-          .bindMemory(to: Int8.self, capacity: dataByteCount)
-        let status = TF_NewStatus()
-        for (cString, tfEncodedSize) in zip(cStrings, tfEncodedSizes) {
-          _ = cString.withUnsafeBufferPointer { buffer in
-            TF_StringEncode(
-              buffer.baseAddress, buffer.count, dataAddr, tfEncodedSize, status)
+
+    // Note: `TENSORFLOW_MASTER` changes below are necessary for the new TensorFlow ABI-stable
+    // unified string tensor design.
+    #if TENSORFLOW_MASTER
```
Comment (dan-zheng) [inline Sources/TensorFlow/Core/StringTensor.swift:52]: Note: Swift for TensorFlow is currently pinned to TensorFlow tag [r2.2](https://github.com/tensorflow/swift-apis/blob/b71c024ff1935e0552b3e44107c647fb6ed24c9c/CMakeLists.txt), so we'll need to add some `#if TENSORFLOW_MASTER` flags to selectively enable the new string tensor behavior until we update the TensorFlow version.

**Comment by dan-zheng** [comment]
@gharibian: could you please review these changes?

**Comment by gharibian** [comment]
> @gharibian: could you please review these changes?

They look good. Let me coordinate with Goldie on versioning (2.3 branch cut was 2 days ago) and get approval from tensorflow-apis on the main CL before moving ahead with this change.

**Comment by gharibian** [comment]
> > @gharibian: could you please review these changes?
> 
> They look good. Let me coordinate with Goldie on versioning (2.3 branch cut was 2 days ago) and get approval from tensorflow-apis on the main CL before moving ahead with this change.

With gTAP and approvals complete, I will be attempting to submit the main CL shortly (the earliest official release will be 2.4).  Feel free to merge this PR as you see fit.

**Comment by dan-zheng** [comment]
Okay, thanks! Merging now.

**Pull request merged**